pkg/runtime/pubsub/bulkpub.go

```diff
@@ -0,0 +1,132 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"context"
+	"errors"
+
+	"golang.org/x/sync/errgroup"
+
+	contribPubsub "github.com/dapr/components-contrib/pubsub"
+	"github.com/dapr/dapr/utils"
+)
+
+const (
+	bulkPublishSeriallyKey       string = "bulkPublishSerially"
+	bulkPublishMaxConcurrencyKey string = "bulkPublishMaxConcurrency"
```

Comment:
Reviewers : for enabling default scenario where if single Publish should be called, should it be called in parallel multiple publish calls or in serial and if in parallel what is the parallelism ... 

all of these will be read as request metadata ...